cmd/podman/images/pull.go

```diff
@@ -156,9 +156,21 @@ func imagePull(cmd *cobra.Command, args []string) error {
 		pullOptions.Username = creds.Username
 		pullOptions.Password = creds.Password
 	}
+
+	rawImgName := args[0]
```

Comment:
This looks like the wrong place. What about `podman run` and other places that auto-pull? Non-CLI users? The Docker HTTP API?

Also, an _intentional decision_ about whether such syntax is allowed for already-pulled images needs to happen.

Reply:
Do I need to modify it in  `containers/image`

Reply:
Please read all of https://github.com/containers/podman/issues/6721 including the linked conversations.